packages/mds-stream/kafka/stream-kafka.ts

```diff
@@ -0,0 +1,55 @@
+import { VehicleEvent, Telemetry, Device } from '@mds-core/mds-types'
+import { ProducerStream } from 'node-rdkafka'
+import { Stream } from '../stream-interface'
+import { createWriteStreamWrapper, isStreamReady, killStream } from './helpers'
+
+let eventStream: ProducerStream | undefined
+let telemetryStream: ProducerStream | undefined
+let deviceStream: ProducerStream | undefined
+
+const initialize = () => {
+  eventStream = createWriteStreamWrapper({}, { topic: 'mds.event' })
+
+  telemetryStream = createWriteStreamWrapper({}, { topic: 'mds.telemetry' })
+
+  deviceStream = createWriteStreamWrapper({}, { topic: 'mds.device' })
+}
+
+const writeTelemetry = async (telemetry: Telemetry[]) => {
+  if (isStreamReady(telemetryStream)) {
+    /*
```

Comment:
I would like to hear more about this bug

Reply:
Let's hop on tandem later, it's such a wacky bug man. Basically what happens, is `isWriteStreamReady` enforces that the `stream` variable passed in is a `ProducerStream` if the check passes (typeguarding it). In any logic that's subsequent, it will then have scoped it down from being `ProducerStream | undefined`, at least, that's what you'd _think_. If you call the scoped down variable in a map, it loses the down-scoping, and returns to being a `ProducerStream | undefined` again, which makes _no sense_.